src/utils/middleware/checkIP.ts

```diff
@@ -23,10 +24,16 @@ export default function checkIP(
     // when the boolean flag "trust proxy" is enabled
     // express takes this from the leftmost value
     // contained in the x-forwarded-for header
-    const errorOrIPString = IPString.decode(req.ip);
+    const errorOrIPString = IPString.decode(req.ip).alt(
+      IPString.decode(req.connection.remoteAddress)
+    );
 
     if (isLeft(errorOrIPString)) {
-      log.error(`Bad request: ${errorOrIPString.value}.`);
+      log.error(
+        `Cannot decode source IP: (req.ip=${req.ip},remoteAddress=${
+          req.connection.remoteAddress
+        },error=${readableReport(errorOrIPString.value)}.`
+      );
       res.status(400).send("Bad request");
```

Comment:
Should we put a message that makes clear the bad request is happening because of ip resolution?

Reply:
@balanza nope, I don't think it would be necessary